### PR TITLE
New package: eiskaltdcpp-qt-2.2.10

### PR DESCRIPTION
**Pull request by samsky72**

### CODE REVIEW

File: srcpkgs/eiskaltdcpp-qt/template [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+revision=1
+wrksrc="eiskaltdcpp-${version}"
+build_style=cmake
+configure_args="-DUSE_MINIUPNP=ON"
+hostmakedepends="pkg-config"
+makedepends="bzip2-devel zlib-devel libressl-devel libidn-devel boost-devel miniupnpc-devel qt-devel"
+short_desc="EiskaltDC++ is a cross-platform Direct Connect and ADC clent"
+maintainer="samsky72 <samsky72@gmail.com>"
+license="GPL-3"
+homepage="https://sourceforge.net/projects/eisklatdcpp"
```
Comment (Johnnynator) [inline srcpkgs/eiskaltdcpp-qt/template]: small type, it shoudl be eisk_al_tcpp and not eisk_la_tcpp.

**Comment by Johnnynator** [comment]
For Cross building you can add
```
if [ -n "$CROSS_BUILD" ]; then
        hostmakedepends+="qt-devel"
fi
```

**Comment by Johnnynator** [comment]
Eiskaltdcpp misses to link against pthread if cross compiling. 
For musl it needs to link against libintl which is provided by gettext-libs (glibc has the needed functionality itself).

```
case $XBPS_TARGET_MACHINE in
        *-musl) makedepends+=" gettext-libs"
                configure_args+=' -DCMAKE_EXE_LINKER_FLAGS="-lintl"';;
        *)      configure_args+=' -DCMAKE_EXE_LINKER_FLAGS="-lpthread"';;
esac
```

musl.patch

```
--- Util.h      2018-01-06 14:36:06.351322188 +0100
+++ dcpp/Util.h 2018-01-06 14:38:02.590360160 +0100
@@ -270,7 +270,7 @@
 #ifdef _WIN32
         return _atoi64(aString.c_str());
 #else
-    #ifndef __HAIKU__
+    #if !defined(__HAIKU__) && defined(__GLIBC__)
         return strtoq(aString.c_str(), (char **)NULL, 10);
     #else
         return strtoll(aString.c_str(), (char **)NULL, 10);
--- main.cpp    2018-01-06 14:46:53.390278957 +0100
+++ eiskaltdcpp-qt/src/main.cpp 2018-01-06 14:52:50.727764605 +0100
@@ -72,7 +72,7 @@
 #if !defined(Q_WS_WIN)
 #include <unistd.h>
 #include <signal.h>
-#if !defined (__HAIKU__)
+#if !defined (__HAIKU__) && defined(__GLIBC__)
 #include <execinfo.h>
 #ifdef ENABLE_STACKTRACE
@@ -127,7 +127,7 @@
         return 0;
     }
-#if !defined (Q_WS_WIN) && !defined (__HAIKU__)
+#if !defined (Q_WS_WIN) && !defined (__HAIKU__) && defined (__GLIBC__)
     installHandlers();
 #endif
```

**Comment by samsky72** [comment]
@Johnnynator Thank you very much!

**Pull request closed without merging**